File: solutions/Maximum-Subarray/Maximum-Subarray.py


class Solution:
    # @param A, a list of integers
    # @return an integer
    def maxSubArray(self, A):
        tempans=A[0]
        ans=A[0]
        for i in range(1,len(A)):
            if tempans>0:
                tempans+=A[i]
            else:
                tempans=A[i]
            ans=max(ans,tempans)
        return ans
 

File: solutions/Maximum-Subarray/Maximum-Subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
  let tempAns = nums[0];
  let ans = tempAns;
  for (let i = 1; i < nums.length; i++) {
    tempAns = tempAns > 0 ? tempAns + nums[i] : nums[i];
    ans = Math.max(ans, tempAns);
  }
  return ans;
};